stream/stream_file.c

```diff
@@ -311,7 +311,7 @@ static int open_f(stream_t *stream, const struct stream_open_args *args)
 #endif
         if (is_fdclose)
             p->close = true;
-    } else if (!strict_fs && !strcmp(filename, "-")) {
+    } else if (!strict_fs && !stream->is_local_fs) {
```

Comment:
this is needlessly confusing
I'd just keep the `stream->is_local_file = true` and set it to false here

Reply:
if we do it that way we need to set it to false in the `fd://` block as well, otherwise `mpv fd://1 --autocreate-playlist=filter` then pressing ctrl+c will result in a segfault. I can do that but I don't really find it simpler